pyanaconda/rescue.py

```diff
@@ -251,41 +240,25 @@ def mount_root(self, root):
         self.status = RescueModeStatus.MOUNTED
         return True
 
-    @property
-    def luks_devices_states(self):
-        """List of objects representing LUKS devices and their state."""
-        if self._luks_devices_states is None:
-            ldevs = [dev for dev in self._storage.devices if dev.format.type == "luks"]
-            self._luks_devices_states = [EncryptedDeviceState(dev) for dev in ldevs]
-        return self._luks_devices_states
-
     def get_locked_device_names(self):
         """List of names of unlocked LUKS devices."""
-        device_names = [device_state.device.name for device_state
-                        in self.luks_devices_states if device_state.locked]
-        return device_names
+        device_names = []
 
-    def _find_device_state(self, device_name):
-        for device_state in self.luks_devices_states:
-            if device_state.device.name == device_name:
-                return device_state
-        return None
+        for device_name in self._device_tree_proxy.GetDevices():
+            format_data = DeviceFormatData.from_structure(
+                self._device_tree_proxy.GetFormatData(device_name)
+            )
 
-    def unlock_device(self, device_name, passphrase):
-        """Unlocks LUKS device."""
-        device_state = self._find_device_state(device_name)
-        if device_state is None:
-            # TODO: raise an exception?
-            log.error("Can't find device to unlock %s", device_name)
-            return False
+            if not format_data.type == "luks":
+                continue
 
-        device = device_state.device
-        unlocked = unlock_device(self._storage, device, passphrase)
+            device_names.append(device_name)
 
-        if unlocked:
-            device_state.set_unlocked(passphrase)
+        return device_names
```

Comment:
I'm confused. Every device what does not have format type "luks" is a locked device? What am I missing? 

Also why is the method name ``get_locked_device_names`` and method description ``List of names of unlocked LUKS devices``?

Reply:
Yes, because we don't support devices unlocked outside of Anaconda.

The docstring was updated.